README.md

```diff
@@ -1,2 +1,66 @@
-# test-things
-Collection of Things that can be used for testing different IoT protocols, security mechanisms and interaction styles
+# Device Sandbox
+
+Collection of IoT device simulators that can be used for testing and exploration purposes of different protocols and other Web of Things mechanisms.
+The devices are implemented via various programming languages and frameworks.
+The protocols you can currently test are:
+
+- HTTP
+- CoAP
+- MQTT
+- Modbus
+
+## Dependencies
+
+The project has several dependencies. Currently `JavaScript` and `Python` is used for simulating different devices. Every device has its own dependencies and they should be handled elegantly. For that reason, `Node.js` is used for JS code and `poetry` is used for Python code to run the scripts and handle the dependencies.
+
+## Testing
+
+For testing JavaScript Testing Framework `mocha` is used. Therefore, the tests are written in JavaScript.
+Every Thing should have its Thing Model and Thing Description validation test. 
+Thing Model validation test should be put under Thing's main directory and named as `tm.test.json`.
+Thing Description validation test should be put under protocol and programming language/framework's test directory and named as `td.test.json`.
+For Thing Description validation test, the device should boot up and to understand the device booted up without any error, a message `"ThingIsReady"` is expected to be prompted to the console by the device.
+
+## Adding a new Thing
+
+If you are going to add a different protocol for an existing Thing:
+1. Create a directory such as `<your_protocol>/<your_programming_language/your_framework>/`.
+2. Create your project files and write your code inside this directory.
+3. If you are adding a new programming language, please use a tidy dependency management tool for the programming language. Otherwise use already used tools and frameworks not to overcomplicate the project.
+4. Create your `test/` directory under your Thing's directory and add your test files there.  
+
+If you are going to add a different programming language/framework for an existing protocol:
+1. Create a directory such as `<existing_protocol>/<your_programming_language/your_framework>/`.
```

Comment:
same as before.. I think it should be under the thing directory

Reply:
Fixed at [156d7be](https://github.com/eclipse-thingweb/test-things/pull/1/commits/156d7be2cc5b36e9d31e612a4f83c4abedef38fc)